### PR TITLE
Alloy-Mixin: allow k8s cluster and alloy cluster disable, add logs dashboard

### PR DESCRIPTION
**Pull request by gaantunes**
#### PR Description

Introducing the following changes:

- including a config file to the mixin, which allows you to:
  - disable dashboards, panels and alerts that are targeted on monitoring alloy instances being run in cluster mode.
  - disable the inclusion of k8s cluster targeted labels on dashboards and alerts.
  - change the dashboard tags
- including a logs dashboard
- including a `job` label everywhere as a grouper when `cluster` and `namespace` are disabled, to avoid metric conflicts - like `up` metric

#### Notes to the Reviewer

Let me know if this needs changelog updates or if there are any docs.

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Review by rfratto — COMMENTED** [review]
Thanks! This is a bit hard to review for concrete differences at the moment given how many things changed, but I left a few comments to help reduce the number of differences so I can give a more thoughtful second pass.

File: operations/alloy-mixin/dashboards/utils/dashboard.jsonnet [hunk 1]
```diff
@@ -77,6 +77,8 @@
     datasource: '${datasource}',
     refresh: 2,
     sort: 2,
+    allValue: '.*',
+    includeAll: true,
```
[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/utils/dashboard.jsonnet:81]: That is to allow queries to work when the target labels are not present.

File: operations/alloy-mixin/dashboards/resources.libsonnet [hunk 2]
```diff
@@ -190,9 +194,9 @@ local stackedPanelMixin = {
         panel.withPosition({ x: 12, y: 16, w: 12, h: 8 }) +
         panel.withQueries([
           panel.newQuery(
-            expr=|||
```
[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/resources.libsonnet:193]: Yes, you are right. I will change it to use that.

File: operations/alloy-mixin/dashboards/opentelemetry.libsonnet [hunk 3]
```diff
@@ -15,21 +15,25 @@ local stackedPanelMixin = {
 };
 
 {
+  local templateVariables = 
+    if $._config.enableK8sCluster then
+      [
+        dashboard.newMultiTemplateVariable('job', 'label_values(alloy_component_controller_running_components, job)'),
+        dashboard.newTemplateVariable('cluster', 'label_values(alloy_component_controller_running_components{job=~"$job"}, cluster)'),
+        dashboard.newTemplateVariable('namespace', 'label_values(alloy_component_controller_running_components{job=~"$job", cluster=~"$cluster"}, namespace)'),
+        dashboard.newMultiTemplateVariable('instance', 'label_values(alloy_component_controller_running_components{job=~"$job", cluster=~"$cluster", namespace=~"$namespace"}, instance)'),
```
[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/opentelemetry.libsonnet:24]: Overall we treat job as the root level label, since integration have a specific job name. I can change the order here for when k8s cluster is enabled, though.

File: operations/alloy-mixin/dashboards/alloy-logs.libsonnet [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+{
+  grafanaDashboards+:
+    if $._config.enableLokiLogs then {
+      local alloyLogs =
+        logsDashboard.new(
+          'Alloy logs overview',
+          datasourceName='loki_datasource',
+          datasourceRegex='',
+          filterSelector=$._config.filterSelector,
+          labels=['cluster', 'namespace', 'instance', 'level'],
```
[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/alloy-logs.libsonnet:12]: You are right, forgot to add the conditional here.

File: operations/alloy-mixin/dashboards/prometheus.libsonnet [hunk 5]
```diff
@@ -2,428 +2,425 @@ local dashboard = import './utils/dashboard.jsonnet';
...
+      defaults+: {
+        custom+: {
+          fillOpacity: 20,
+          gradientMode: 'hue',
+          stacking: { mode: 'normal' },
+        },
       },
     },
   },
-};
```
[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/prometheus.libsonnet:15]: I had some syntax problems keeping it out of the object, so just moved it in since there is no impact on the output.

[the remark this answers is not in the document]
Comment (rfratto) [reply operations/alloy-mixin/dashboards/prometheus.libsonnet:15]: OK 👍 I mainly brought this up because of the diffs being hard to follow. If everything else is addressed and it doesn't impact the diffs too much we can leave it, otherwise I'll want to revisit this and figure out what the syntax issues were.

[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/prometheus.libsonnet:15]: For this variable specifically it is not needed, but to keep consistency I just moved everthing.

File: operations/alloy-mixin/config.libsonnet [hunk 6]
```diff
@@ -1,10 +1,10 @@
 {
     _config+:: {
-        enableK8sCluster: true,
+        enableK8sCluster: false,
```
Comment (rfratto) [inline operations/alloy-mixin/config.libsonnet:3]: Did you mean to change this default? It was true in the previous commit

File: operations/alloy-mixin/alerts.libsonnet [hunk 7]
```diff
@@ -1,9 +1,20 @@
...
+        clusterAlerts.newAlloyClusterAlertsGroup($._config.enableK8sCluster),
+        controllerAlerts.newControllerAlertsGroup($._config.enableK8sCluster),
+        openTelemetryAlerts.newOpenTelemetryAlertsGroup($._config.enableK8sCluster),
+      ]
+      else         
+      [
+        controllerAlerts.newControllerAlertsGroup($._config.enableK8sCluster),
+        openTelemetryAlerts.newOpenTelemetryAlertsGroup($._config.enableK8sCluster)
+      ],
   },
```
Comment (rfratto) [inline operations/alloy-mixin/alerts.libsonnet:19]: Maintenance wise, I'm worried about having two arrays where someone might forget to put a new alert group on both. Can we change this to be additive similarly to how `dashboards.libsonnet` has "otherDashboards" and "alloyClusterDashboards"?

File: operations/alloy-mixin/alerts/clustering.libsonnet [hunk 8]
```diff
@@ -1,67 +1,97 @@
...
+            on (cluster, namespace, job) group_left
+            count by (cluster, namespace, job) (cluster_node_info)
+          |||
+          else
+          |||
+            sum without (state) (cluster_node_peers) !=
+            on (job) group_left
+            count by (job) (cluster_node_info)
+          |||
+          ,
```
Comment (rfratto) [inline operations/alloy-mixin/alerts/clustering.libsonnet:37]: The formatting here seems a bit non-standard:

```suggestion
          if enableK8sCluster then |||
            sum without (state) (cluster_node_peers) !=
            on (cluster, namespace, job) group_left
            count by (cluster, namespace, job) (cluster_node_info)
          ||| else |||
            sum without (state) (cluster_node_peers) !=
            on (job) group_left
            count by (job) (cluster_node_info)
          |||,
```

File: operations/alloy-mixin/dashboards/alloy-logs.libsonnet [hunk 9]
```diff
@@ -0,0 +1,35 @@
+local g = import 'github.com/grafana/grafonnet/gen/grafonnet-v10.0.0/main.libsonnet';
+local logsDashboard = import 'github.com/grafana/jsonnet-libs/logs-lib/logs/main.libsonnet';
+
+{
+
+  local labels = if $._config.enableK8sCluster then ['cluster', 'namespace', 'job', 'instance', 'level'] else ['job', 'instance', 'level'],
+
+  grafanaDashboards+:
```
Comment (rfratto) [inline operations/alloy-mixin/dashboards/alloy-logs.libsonnet:8]: This doesn't seem to follow the pattern that other dashboards do; this is the only one with a `grafanaDashboards` top-level key. Can we change this to be consistent with the other dashboards? 

That would also mean moving the logic for whether to enable the dashboard outside of the definition of the dashboard itself, which would be more consistent with how the clustering dashboards are made optional.

File: operations/alloy-mixin/dashboards/cluster-overview.libsonnet [hunk 10]
```diff
@@ -97,7 +102,7 @@ local cluster_node_filename = 'alloy-cluster-node.json';
                     {
                       targetBlank: false,
                       title: 'Detail dashboard for node',
-                      url: '/d/%(uid)s/alloy-cluster-node?var-instance=${__data.fields.instance}&var-datasource=${datasource}&var-loki_datasource=${loki_datasource}&var-cluster=${cluster}&var-namespace=${namespace}' % { uid: std.md5(cluster_node_filename) },
+                      url: '/d/%(uid)s/alloy-cluster-node?var-instance=${__data.fields.instance}&var-datasource=${datasource}&var-loki_datasource=${loki_datasource}&var-job=${job}&var-cluster=${cluster}&var-namespace=${namespace}' % { uid: std.md5(cluster_node_filename) },
```
Comment (rfratto) [inline operations/alloy-mixin/dashboards/cluster-overview.libsonnet:105]: Nice find here, this would've been challenging to debug :)

File: operations/alloy-mixin/dashboards/utils/dashboard.jsonnet (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/utils/dashboard.jsonnet:81]: I understand your concerns. This is mostly targeted on `cluster` and `namespace` labels. Since we do not have the ability to use this `enableK8sCluster` variable when installing the mixin, these variables are always going to be present, even though the user is running a local/non cluster environment.

In this scenario, if we do not allow `all`, no data will be displayed on the dashboards. So it is a hard trade off.

**Review by rfratto — COMMENTED** [review]
Thanks for the changes!

File: operations/alloy-mixin/dashboards/utils/dashboard.jsonnet (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/utils/dashboard.jsonnet:81]: Just tested it without `all` option enabled for `cluster` and `namespace`, and it seems to work when this label is not present. So it should be OK to get back to the previous state.

File: operations/alloy-mixin/dashboards/alloy-logs.libsonnet (hunk 9 above)
Comment (gaantunes) [reply operations/alloy-mixin/dashboards/alloy-logs.libsonnet:8]: That is because the library I am using expects a `grafanaDashboards` variable.
I also don't like how it looks different, but since this is a totally different library which is synced with the latest version of grafonnet, I believe it is an acceptable trade off.

**Review by rfratto — APPROVED** [review]
No other comments from me, nice work making this configurable!
